wire-library/wire-compiler/src/main/java/com/squareup/wire/WireLogger.kt

```diff
@@ -33,16 +33,26 @@ interface WireLogger {
    * @param qualifiedName is the file path when generating a `.proto` file, the type or service
    *   name prefixed with its package name when generating a `.java` or `.kt` file, and the type
    *   name when generating a `.swift` file.
+   * @param targetName is used to identify the concerned target. For
+   * [JavaTarget][com.squareup.wire.schema.JavaTarget], the name will be "Java". For
+   * [KotlinTarget][com.squareup.wire.schema.KotlinTarget], the name will be "Kotlin". For
+   * [SwiftTarget][com.squareup.wire.schema.SwiftTarget], the name will be "Swift". For
+   * [ProtoTarget][com.squareup.wire.schema.ProtoTarget], the name will be "Proto".
```

Comment:
Would you be open to a change that broke `Java`, `Kotlin`, `Swift` and `Proto` out into an enum?

Reply:
(or sealed classes, for that sweet-sweet exhaustive `when`s)

Reply:
I didn't wanna block a consumer from putting whatever they wanted here. Like custom handler should be able to use the logger and call those methods too.